### PR TITLE
Mccalluc/dev search

### LINKED ISSUE #825: dev-tools search to see which index version is up
We shouldn't deploy to prod until the index is up to date. Add a search for developer data which would show that as a facet.

### PR DESCRIPTION
**Pull request by mccalluc**
Add a dev-search to help us find examples of documents that exercise particular parts of the UI. Fixes #825

### CODE REVIEW

File: context/app/static/js/components/Search/DevSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import Typography from '@material-ui/core/Typography';
+import { ExistsQuery, BoolMustNot } from 'searchkit';
+import SearchWrapper from './SearchWrapper';
+import { readCookie } from '../../helpers/functions';
+import './Search.scss';
```
Comment (john-conroy) [inline context/app/static/js/components/Search/DevSearch.jsx:6]:
```suggestion
import Typography from '@material-ui/core/Typography';
import { ExistsQuery, BoolMustNot } from 'searchkit';

import { readCookie } from 'helpers/functions';
import SearchWrapper from './SearchWrapper';
import './Search.scss';
```

File: context/app/static/js/components/Search/DevSearch.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { field, filter, checkboxFilter } from './utils';
+import { lastModifiedSort, sizeSort } from './config';
+
+const nexus_token = readCookie('nexus_token');
+const httpHeaders = nexus_token
+  ? {
+      Authorization: `Bearer ${nexus_token}`,
+    }
+  : {};
+const searchProps = {
```
Comment (john-conroy) [inline context/app/static/js/components/Search/DevSearch.jsx:17]:
```suggestion
  : {};

const searchProps = {
```

File: context/app/static/js/components/Search/DevSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    checkboxFilter('no_files', 'No files?', BoolMustNot(ExistsQuery('files'))),
+  ],
+  sortOptions: lastModifiedSort.concat(sizeSort),
+  queryFields: ['everything'],
+  isLoggedIn: Boolean(nexus_token),
+};
+
+function DevSearch(props) {
+  const { elasticsearchEndpoint } = props;
+  const allProps = Object.assign(searchProps, { apiUrl: elasticsearchEndpoint });
```
Comment (john-conroy) [inline context/app/static/js/components/Search/DevSearch.jsx:52]:
```suggestion
  const allProps = {...searchProps, apiUrl: elasticsearchEndpoint};
```
We can use spread here?

File: context/app/static/js/components/Search/Search.jsx [hunk 4]
```diff
@@ -1,21 +1,15 @@
 import React from 'react';
 import Typography from '@material-ui/core/Typography';
-import { ExistsQuery, BoolMustNot } from 'searchkit';
 import SearchWrapper from './SearchWrapper';
 import { readCookie } from '../../helpers/functions';
```
Comment (john-conroy) [inline context/app/static/js/components/Search/Search.jsx:4]:
```suggestion
import { readCookie } from 'helpers/functions';
import SearchWrapper from './SearchWrapper';
```

File: context/app/static/js/components/Search/Search.jsx [hunk 5]
```diff
@@ -1,21 +1,15 @@
 import React from 'react';
 import Typography from '@material-ui/core/Typography';
```
Comment (john-conroy) [inline context/app/static/js/components/Search/Search.jsx:2]:
```suggestion
import Typography from '@material-ui/core/Typography';

```

File: context/app/static/js/components/Search/Search.jsx [hunk 6]
```diff
@@ -1,21 +1,15 @@
...
 import SearchWrapper from './SearchWrapper';
 import { readCookie } from '../../helpers/functions';
 import './Search.scss';
-import { donorConfig, sampleConfig, datasetConfig } from './config';
+import { donorConfig, sampleConfig, datasetConfig, lastModifiedSort } from './config';
 // eslint-disable-next-line import/named
-import { filter, checkboxFilter } from './utils';
+import { filter } from './utils';
 import AncestorNote from './AncestorNote';
 import LookupEntity from '../../helpers/LookupEntity';
```
Comment (john-conroy) [inline context/app/static/js/components/Search/Search.jsx:10]: Perhaps for another PR, but could we make this a hook? We can also change the import path to `helpers/LookupEntity`, the import statement will have to be moved below the import of readCookie due eslint import order issues.

Comment (mccalluc) [reply context/app/static/js/components/Search/Search.jsx:11]: Filed #860, and I'll change the path here.

**Comment by mccalluc** [comment]
> I assume its okay for knowledgable users to access dev-search?

Yep, this page doesn't reveal any sensitive details.

> In a future PR can we begin to ogranize Search? I think it's becoming slightly untidy with the different files/file types being added.

Sure, however you see fit: Search and DevSearch share a lot, so it makes sense to me to keep those extra files close, rather than moving them to helpers, but that could be the wrong call.

**Pull request merged**